### DISCUSSION TITLE
Summer Communication and The Organization of Development

### DISCUSSION THREAD

**Discussion by PetalCat** [PSDTools/app#19] [General]
We need to find a way to effectively communicate and organize things that need to be done with the [app](https://github.com/PSDTools/app)...

We should also create a timeline that leaves plenty of space for:
- Oh crap this doesn't work, or if
- We need to rethink some things.

I think we can do most of the communication and organization through GitHub but we should probably have a way to verbally talk even if it is only every once and a while over
- Zoom,
- Google Meet,
- etc.

I do still think it would be really cool to set up a live IDE with Replit but Replit and Flutter don't like to play nice.[^1][^2]

[^1]: I did see somewhere that someone got it to work
[^2]: It would also be nice if it worked just for my sake, as I move between computers often, and all I would ever need is a web browser; if it doesn't work, though, I am cool with just using Git.

**Comment by lishaduck** [comment]
- GitHub has a timeline option under projects.
  - However, you need issues for every item and follow scrum,[^1] somewhat.
- We can communicate via GitHub for the most part, unless you have a better option.
  - I think checking in semiweekly-ish over Zoom/Meet sounds reasonable.
  - However, checking in biweekly-ish in person would probably be nice.
- If we need a live IDE, we can just set up Live Share.[^2]
  - If you practice working with Git, it should be fine.
  - You'll just need to get to a stopping point[^3] each time where you can commit to a branch.

[^1]: We might as well be agile, though.
[^2]: Audio calling is gone now, though.
[^3]: Not completely done, but where there's not a syntax error.

**Comment by PetalCat** [comment]
I think we should shoot for having the minimum viable product by the end of July. That would give us 4-5 weeks to polish and release ~(School admin are slloooooww)~.[^1] We don't need all the bells and whistles by then but enough to be functional enough to work with admin then push polish and feature updates for those 4 or so weeks and after release the occasional patch or feature as necessary. I know this squashes us really tight but I think we can get a teacher, student, and admin flavor and the backend set up. I am pretty sure the hardest part will be PowerSchool authentication and the Python server (dealing with the auth).

[^1]: Censored by the _"Commit"y to Prevent the Slandering of Government Officials_ in its famous overreach of power 🤣

**Reply by PetalCat** [reply]
Lmao the footnote

**Reply by lishaduck** [reply]
Thank you.

**Comment by lishaduck** [comment]
Is our current setup sufficient, @ParkerH27 / @PetalCat?

**Comment by PetalCat** [comment]
Yep... We should probably talk more often but besides that we are fine
